config/config.go

```diff
@@ -268,6 +276,25 @@ func mergeConfigWithIncludedConfig(config *TerragruntConfig, includedConfig *Ter
 	return includedConfig, nil
 }
 
+// Merge the extra arguments prioritizing those defined in the childExtraArgs
+func mergeExtraArgs(terragruntOptions *options.TerragruntOptions, childExtraArgs []TerraformExtraArguments, parentExtraArgs *[]TerraformExtraArguments) {
+	result := *parentExtraArgs
+addExtra:
+	for _, child := range childExtraArgs {
+		for i, parent := range result {
+			if parent.Name == child.Name {
+				terragruntOptions.Logger.Printf("extra_arguments '%v' from child overriding parent", child.Name)
+				// For extra args, we want to keep the values specified in the child and put them after
+				// the parent ones, so if we encounter a duplicate, we just overwrite it.
+				result[i] = child
+				continue addExtra
```

Comment:
Ok, looks like the _last_ tfvars file on the command line containing a variable "wins" in the case where the variable is defined in multiple files.  I tested this out in terraform.

So, I think the current implementation is "functionally correct", in that it always adds child extra args at the end.  Therefore, terragrunt will put the child's extra_arguments` after the parent's `extra_arguments` on the terraform command line.

I have clarified this behavior in the documentation.

Let me know what you think.